src/test_utils.rs

```diff
@@ -43,21 +43,35 @@ pub fn get_lines_content<P: AsRef<Path>>(path: P) -> BTreeSet<String> {
     set
 }
 
+fn get_files_to_compare<P>(output_dir: &P, files_to_check: Option<Vec<&str>>) -> Vec<String>
+where
+    P: AsRef<Path>,
+{
+    match files_to_check {
+        None => fs::read_dir(output_dir.as_ref())
+            .unwrap()
+            .map(|f| f.unwrap().file_name().into_string().unwrap())
+            .collect(),
+        Some(v) => v.iter().map(|&f| f.to_string()).collect(),
+    }
+}
+
 pub fn compare_output_dir_with_expected<P: AsRef<Path>>(
     output_dir: &P,
     files_to_check: Option<Vec<&str>>,
     work_dir_expected: &str,
 ) {
-    let output_dir = output_dir.as_ref();
-    let files: Vec<String> = match files_to_check {
-        None => fs::read_dir(output_dir)
-            .unwrap()
-            .map(|f| f.unwrap().file_name().into_string().unwrap())
-            .collect(),
-        Some(v) => v.iter().map(|&f| f.to_string()).collect(),
-    };
+    compare_output_dir_with_expected_lines(output_dir, files_to_check, work_dir_expected);
+}
+
+pub fn compare_output_dir_with_expected_lines<P: AsRef<Path>>(
```

Comment:
I renamed the 2 new test functions:
- from `compare_output_dir_with_expected_csv` into `compare_output_dir_with_expected_lines`
- from `compare_output_dir_with_expected_xml` into `compare_output_dir_with_expected_content`

See f3d228f80708040924d80493938566f783a7ae2d.